src/__swaps__/screens/Swap/components/GasButton.tsx

```diff
@@ -87,26 +90,36 @@ const GasMenu = ({ backToReview = false, children }: { backToReview?: boolean; c
   const metereologySuggestions = useMeteorologySuggestions({ chainId });
   const customGasSettings = useCustomGasSettings(chainId);
 
-  const menuOptions = useMemo(() => [...keys(metereologySuggestions.data), 'custom'] as const, [metereologySuggestions.data]);
+  const menuOptions = useMemo(() => [...keys(metereologySuggestions.data), GasSpeed.CUSTOM] as GasSpeed[], [metereologySuggestions.data]);
 
   const handlePressSpeedOption = useCallback(
     (selectedGasSpeed: GasSpeed) => {
-      setSelectedGasSpeed(chainId, selectedGasSpeed);
+      const { inputAsset, outputAsset, quote } = swapsStore.getState();
+      analyticsV2.track(analyticsV2.event.swapsChangedSelectedGasSpeed, {
```

Comment:
I think we can just track this when a swap is submitted (the selected speed, or custom, on submit)